text/0000-language-semver.md

```diff
@@ -0,0 +1,494 @@
+    running code are obviously much more concerning than minor corner
+    cases. There is somewhat in tension with the other factors: if
+    there is, for example, a widely deployed vulnerability, fixing
+    that vulnerability is important, but it will also cause a larger
+    disruption.
+- How many crates on `crates.io` are affected?
+  - This is a general proxy for the overall impact (since of course
+    there will always be private crates that are not part of
+    crates.io).
+- Were particularly vital or widely used crates affected?
+  - This could indicate that the impact will be wider than the raw
+    number would suggest.
+- Does the change silently change the result of running the program,
+  or simply cause additional compilation failures?
+  - The latter, while frustrating, are easier to diagnose.
+- What changes are needed to get code compiling again? Are those
+  changes obvious from the error message?
+  - The more cryptic the error, the more frustrating it is when
+    compilation fails.
+    
+#### What is a "compiler bug" or "soundness change"?
+
+In the absence of a formal spec, it is hard to define precisely what
+constitutes a "compiler bug" or "soundness change" (see also the
+section below on underspecified parts of the language). The obvious
+cases are soundness violations in a rather strict sense:
+
+- Cases where the user is able to produce Undefined Behavior (UB)
+  purely from safe code.
+- Cases where the user is able to produce UB using standard library
+  APIs or other unsafe code that "should work".
+    
+However, there are other kinds of type-system inconsistencies that
+might be worth fixing, even if they cannot lead directly to UB.  Bugs
+in the coherence system that permit uncontrolled overlap between impls
+are one example. Another example might be inference failures that
+cause code to compile which should not (because ambiguities
+exist). Finally, there is a list below of areas of the language which
+are generally considered underspecified.
+
+We expect that there will be cases that fall on a grey line betwen bug
+and expected behavior, and discussion will be needed to determine
+where it falls. The recent conflict between `Rc` and scoped threads is
+an example of such a discusison: it was clear that both APIs could not
+be legal, but not clear which one was at fault. The results of these
+discussions will feed into the Rust spec as it is developed.
+    
+#### Opting out
+
+In some cases, it may be useful to permit users to opt out of new type
+rules. The intention is that this "opt out" is used as a temporary
+crutch to make it easy to get the code up and running. Typically this
+opt out will thus be removed in a later release. But in some cases,
+particularly those cases where the severity of the problem is
+relatively small, it could be an option to leave the "opt out"
+mechanism in place permanently. In either case, use of the "opt out"
+API would trigger the deprecation lint.
```

Comment:
@pnkfelix 

> The "Unresolved questions" section makes it sound like you previously established that the "opt out" mechanism would be some sort of attribute, but I do not see that specifically spelled out here 

Yes, I just assumed "opt out" would be an attribute because it allows us to target the opt-out at very fine granularity (e.g., exempt this impl from the coherence rules, but not the rest).